symphony-app-authentication/symphony-app-authentication-filter/src/test/java/org/symphonyoss/symphony/apps/authentication/factories/ServicesInfoProviderFactoryTest.java

```diff
@@ -0,0 +1,50 @@
+package org.symphonyoss.symphony.apps.authentication.factories;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import org.junit.Test;
+import org.symphonyoss.symphony.apps.authentication.MockServicesInfoProvider;
+import org.symphonyoss.symphony.apps.authentication.ServicesInfoProvider;
+
+/**
+ * Unit tests for {@link ServicesInfoProviderFactory}
+ *
+ * Created by robson on 09/01/18.
+ */
+public class ServicesInfoProviderFactoryTest {
+
+  private static final String MOCK_POD_URL = "https://mock.test.com/pod";
+
+  private static final String MOCK_SESSION_AUTH_URL = "https://mock-api.test.com/sessionauth";
+
+  private static final ServicesInfoProvider MOCK_PROVIDER =
+      new MockServicesInfoProvider(MOCK_POD_URL, MOCK_SESSION_AUTH_URL);
+
+  private ServicesInfoProviderFactory factory = ServicesInfoProviderFactory.getInstance();
+
+  @Test
+  public void testNullComponent() {
+    try {
+      factory.setComponent(null);
+      fail();
```

Comment:
Is it necessary? On method setComponent, if passed null to method setComponent then should return the exception.

Reply:
I'm following this approach to make sure the exception have been thrown by the component and check the exception message.